frontend/awx/views/jobs/JobOutput/StatusBar.tsx

```diff
@@ -81,27 +133,44 @@ export function HostStatusBar(props: { counts: HostStatusCounts }) {
     },
   };
 
-  const barSegments = Object.keys(hostStatus).map((key) => {
-    const count = counts[key as StatusCountType] || 0;
-    const status = hostStatus[key as StatusCountType];
+  return <StatusBar counts={props.counts} status={hostStatus} />;
+}
+
+interface StatusBarProps<T extends object, K> {
+  counts: T;
+  status: K;
+}
+
+function StatusBar<T extends object, K extends object>(props: StatusBarProps<T, K>) {
+  const { t } = useTranslation();
+  const { counts, status } = props;
+  const noData = Object.keys(counts).length === 0;
+  const totalCounts = Object.values(counts).reduce(
+    (sum: number, count: number) => sum + count,
+    0
+  ) as number;
+
+  const barSegments = Object.keys(status).map((key) => {
+    const count = (counts[key as keyof T] as number) || 0;
+    const jobStatus = status[key as keyof K] as StatusProps;
```

Comment:
I am getting an error locally here.

<img width="416" alt="image" src="https://github.com/ansible/ansible-ui/assets/9053044/6e6e6505-210c-4e6d-b128-274bded118b8">

It seems like we could add default values to the color and label.